main.py

```
@@ -25,5 +25,50 @@ async def build_esp_related_project(project_path: str) -> (str, str):
+    
+    # Build the flash command
+    if port:
+        flash_cmd = f"bash -c 'source {export_script} && idf.py -p {port} flash'"
+    else:
+        flash_cmd = f"bash -c 'source {export_script} && idf.py flash'"
+    
+    returncode, stdout, stderr = await run_command_async(flash_cmd)
+    
+    # Log the flash operation
+    flash_log = f"Flash operation - Return code: {returncode}\nSTDOUT:\n{stdout}\nSTDERR:\n{stderr}"
+    open('mcp-flash.log', 'w+').write(flash_log)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)